src/app-layout/utils/use-drawers.ts

```diff
@@ -95,28 +110,32 @@ export function useDrawers(
   );
   const [drawerSizes, setDrawerSizes] = useState<Record<string, number>>({});
 
-  const onActiveDrawerChange = useStableCallback((newDrawerId: string | undefined) => {
+  function onActiveDrawerResize({ id, size }: { id: string; size: number }) {
+    setDrawerSizes(oldSizes => ({ ...oldSizes, [id]: size }));
+    fireNonCancelableEvent(ownDrawers?.onResize, { id, size });
+  }
+
+  function onActiveDrawerChange(newDrawerId: string | undefined) {
     setActiveDrawerId(newDrawerId);
-    fireNonCancelableEvent(ownDrawers?.onChange, newDrawerId);
-  });
+    if (hasOwnDrawers) {
+      fireNonCancelableEvent(ownDrawers?.onChange, newDrawerId);
+    }
+    if (!toolsProps.toolsHide) {
+      fireNonCancelableEvent(toolsProps.onToolsChange, { open: newDrawerId === TOOLS_DRAWER_ID });
```

Comment:
Removed repeating `onToolsChange` handlers from other places, this is the only one where we keep it to support compatibility between tools and drawers

Reply:
Do we need to check if tools are ignored?

Reply:
We still have "multi drawer with tools mode" for runtime API, this is why the condition is here. See `applyToolsDrawer` function how exactly this works